src/Transport/Topology/Topologies/ForwardingTopology.cs

```diff
@@ -109,6 +110,14 @@ public Func<IDispatchMessages> GetDispatcherFactory()
             return () => container.Resolve<IDispatchMessages>();
         }
 
+        public StartupCheckResult ApplyPreStartupChecks()
+        {
+            var checks = container.ResolveAll<IStartupCheck>();
+            var compositeStartupChecker = new CompositeStartupCheck(checks);
```

Comment:
> You do that by allowing custom topology to implement a pre-startup check.

Do you mean that to add a single pre-startup check, a user is forced to implement from scratch a whole custom topology? Did I get it right?

I don't know the design concepts behind ASB trasport implementation...yet :). If you think that this behaviour is not indispensable, I'll make code easier without using container and building `CompositeStartupCheck` directly.